src/modules/filters/getDataOnFilter.ts

```diff
@@ -0,0 +1,49 @@
+import { Product } from '../../types';
+import data from '../../data/db';
+import { Filter } from '../../types';
+
+export function getDataFilter(filter: Filter) {
+  let result: Array<Product> = [];
+  result = data.filter((el) => {
+    if (
+      (+filter.year[0] <= +el.year &&
+        +el.year <= +filter.year[1] &&
+        +filter.count[0] <= +el.count &&
+        +el.count <= +filter.count[1]) ||
+      (filter.year.length <= 0 && filter.count.length <= 0)
+    )
+      return true;
```

Comment:
Абсолютно то же самое
```suggestion
return (
      (+filter.year[0] <= +el.year &&
        +el.year <= +filter.year[1] &&
        +filter.count[0] <= +el.count &&
        +el.count <= +filter.count[1]) ||
      (filter.year.length <= 0 && filter.count.length <= 0)
    );
```